### DISCUSSION TITLE
Why not use top metal layers for PDN?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD-flow-scripts#2427] [General]
Which layers that can be used in a macro for routing is limited to the top PDN layer.

Is there a reason not to move the PDN to higher metal layers in these files to allow using more routing layers?

https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/edcd265180fc6f2edc04b6743920fc027c56c797/flow/platforms/asap7/openRoad/pdn/BLOCK_grid_strategy.tcl#L32C1-L32C44

https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/edcd265180fc6f2edc04b6743920fc027c56c797/flow/platforms/asap7/openRoad/pdn/BLOCKS_grid_strategy.tcl#L28C1-L28C44

**Comment by maliberty** [comment]
You are trading routability at the block level vs at the top level.  The right tradeoff can be design specific.

**Reply by oharboe** [reply]
Gotit.

So if I dont need or want to route on top of macros and want to use all metal layers for routing, is there a PDN .tcl file I could use?

Or adapt easily(as in with limited skills and much guesswork)?

**Reply by maliberty** [reply]
Just modify a copy of the existing pdn script to change the relevant layers.  It is possible to connect the power grid to the block in planar sense but it requires that the straps line up exactly with the power pins.  It is somewhat easier to leave two layers at the top for power connection from above.

**Reply by oharboe** [reply]
What about all the magic numbers?

Are they not layer specific?

**Reply by maliberty** [reply]
A mix of layer specific and design specific.  The values we give are just reasonable guesses.  Really this is driven by IR drop analysis and human judgement of acceptable results.  Often foundries have recommended grids but asap7 has no foundry.

**Reply by oharboe** [reply]
Thanks! Will give it a go.
